knownprojects_build/sql/dcp_application.sql

```diff
@@ -158,143 +103,86 @@ year_filter as (
 ),
-
---all projects that have text pattern matched to be residential
-text_filter as (
-    SELECT dcp_name
-    FROM dcp_projects
-    WHERE (regexp_replace(
-        array_to_string(
-            ARRAY[dcp_projectbrief,
-                dcp_projectdescription,
-                dcp_projectname], 
-            ' '),
-        '[^a-zA-Z0-9]+', ' ','g'
-        ) like '%DUs%'
-        or regexp_replace(
-            array_to_string(
-                ARRAY[dcp_projectbrief,
-                    dcp_projectdescription,
-                    dcp_projectname], 
-                ' '),
-            '[^a-zA-Z0-9]+', ' ','g'
-            ) like '%MIH%'
-        or regexp_replace(
-            array_to_string(
-                ARRAY[dcp_projectbrief,
-                    dcp_projectdescription,
-                    dcp_projectname], 
-                ' '),
-            '[^a-zA-Z0-9]+', ' ','g'
-        ) ~* 
-        array_to_string(ARRAY[
-            'HUDSON YARDS',
-            'home','family','resid',
-            'appartment','apt','affordable', 
-            'mix-','mixed-', 'dwelling',
-            'living', 'housi'], '|')
-    )
-    AND regexp_replace(
-        array_to_string(
-            ARRAY[dcp_projectbrief,
-                dcp_projectdescription,
-                dcp_projectname],
-            ' '),
-        '[^a-zA-Z0-9]+', ' ','g'
-    ) !~* 
-        array_to_string(ARRAY[
-        'RESIDENTIAL TO COMMERCIAL', 
-        'SINGLE-FAMILY', 'SINGLE FAMILY', 
-        '1-FAMILY', 'ONE FAMILY', 
-        'ONE-FAMILY', '1 FAMILY',
-        'FLOATING', 'TRANSITIONAL', 'FOSTER', 
-        'ILLUMIN', 'RESIDENCE DISTRICT', 
-        'LANDMARKS PRESERVATION COMMISSION',
-        'EXISTING HOME', 'EXISTING HOUSE', 
-        'NUMBER OF BEDS', 'EATING AND DRINKING', 
-        'NO INCREASE', 'ENLARGEMENT', 'NON-RESIDENTIAL', 
-        'LIVINGSTON', 'AMBULATORY', 
-        'APPLICATION FOR PARKING',
-        'CHAIRPERSON CERTIFICATION',
-        'Chair Cert',
-        'ROOFTOP'], '|')
-),
-consolidated_filter as (
-    SELECT dcp_name 
-    FROM text_filter
-    union
-    SELECT dcp_name 
-    FROM resid_units_filter
-),
-applicanttype_filter as (
-	SELECT dcp_name 
-	FROM zap_translated
-	WHERE dcp_applicanttype = 'DCP'
-),
 records_last_kpdb as (
-	SELECT record_id 
+	SELECT record_id as dcp_name
 	FROM dcp_knownprojects
 	WHERE source = 'DCP Application'
 ),
--- records_last_dcp_project as (
--- 	SELECT dcp_name as record_id
--- 	FROM kpdb_2020.dcp_project
--- ),
 records_corr_remove as (
-	SELECT record_id 
-	FROM corrections_main
-	WHERE field = 'remove'
+	SELECT record_id as dcp_name
+	FROM corrections_zap
+	WHERE lower(action) = 'remove'
 ),
 records_corr_add as (
-	SELECT record_id 
-	FROM corrections_main
-	WHERE field = 'add'
+	SELECT record_id as dcp_name
+	FROM corrections_zap
+	WHERE lower(action) = 'add'
+),
+consolidated_add_filter as (
+	/*
+	Add if a record satisfies:
```

Comment:
It can either be option 1, or can be BOTH 2 & 3? Do I understand this correctly?

Reply:
yes, 1 or (2 and 3)